logstash-core/spec/logstash/compiler/compiler_spec.rb

```diff
@@ -0,0 +1,580 @@
+require "spec_helper"
+require "logstash/compiler"
+java_import Java::OrgLogstashConfigIr::DSL
+
+describe LogStash::Compiler do
+  def j
+    Java::OrgLogstashConfigIr::DSL
+  end
+
+  # Static import of these useful enums
+  INPUT = Java::OrgLogstashConfigIr::PluginDefinition::Type::INPUT
+  FILTER = Java::OrgLogstashConfigIr::PluginDefinition::Type::FILTER
+  OUTPUT = Java::OrgLogstashConfigIr::PluginDefinition::Type::OUTPUT
+  CODEC = Java::OrgLogstashConfigIr::PluginDefinition::Type::OUTPUT
+
+  describe "compiling to Pipeline" do
+    subject(:source_file) { "fake_sourcefile" }
+    subject(:compiled) { described_class.compile_pipeline(source, source_file) }
+
+    describe "complex configs" do
+      shared_examples_for "compilable LSCL files" do |path|
+        describe "parsing #{path}" do
+          let(:source) {
+            File.open(path) {|f| f.read }
+          }
+          
+          it "should compile" do
+            expect(compiled).to be_java_kind_of(Java::OrgLogstashConfigIr::Pipeline)
+          end
+          
+          it "should have a hash" do
+            expect(compiled.uniqueHash)
+          end
+        end
+      end
+      
+      Dir.glob(File.join(SUPPORT_DIR, "lscl_configs", "*.conf")).each do |path|
+        it_should_behave_like "compilable LSCL files", path
+      end
+    end
+  end
+
+  describe "compiling imperative" do
+    subject(:source_file) { "fake_sourcefile" }
+    subject(:compiled) { described_class.compile_imperative(source, source_file) }
+
+    describe "an empty file" do
+      let(:source) { "input {} output {}" }
+
+      it "should have an empty input block" do
+        expect(compiled[:input]).to ir_eql(j.noop)
+      end
+
+      it "should have an empty filter block" do
+        expect(compiled[:filter]).to ir_eql(j.noop)
+      end
+
+      it "should have an empty output block" do
+        expect(compiled[:output]).to ir_eql(j.noop)
+      end
+    end
+
+    describe "SourceMetadata" do
+      let(:source) { "input { generator {} } output { }" }
+
+      it "should attach correct source text for components" do
+        expect(compiled[:input].get_meta.getSourceText).to eql("generator {}")
+      end
+    end
+
+    context "plugins" do
+      subject(:c_plugin) { compiled[:input] }
+      let(:source) { "input { #{plugin_source} } " }
+
+      describe "a simple plugin" do
+        let(:plugin_source) { "generator {}" }
+
+        it "should contain the plugin" do
```

Comment:
I agree that this can be improved. 

I think the problem is that sometimes I group tests in a way that doesn't lend itself to english language sentences, but is useful in the actual file. The `context 'plugins'` block that this is nested in (which is responsible for the first word 'plugin' makes sense when reading the file, but makes for a bad test name.

Maybe the solution is to wind up with a sentence like:

`Plugins: A simple plugin declaration should create a plugin IR instance`.

So then I'd have `context 'Plugins: '`. 

WDYT?